config/7.3.0/fapi1part2adv/ig/scripts/groovy/ErrorResponseFilter.groovy

```diff
@@ -0,0 +1,44 @@
+/**
+ * This script is a simple implementation of HTTP Basic Authentication on
+ * server side.
+ * It expects the following arguments:
+ *  - realm: the realm to display when the user-agent prompts for
+ *    username and password if none were provided.
+ *  - username: the expected username
+ *  - password: the expected password
+ */
+SCRIPT_NAME = "[ErrorResponseFilter] - "
+
+logger.debug(SCRIPT_NAME + "Running...")
+
+class ErrorResponse {
+    public String error_description
+    public String error
+}
+
+return next.handle(context, request).thenOnResult {
+    response -> {
+        if(response.status.isClientError()){
+            logger.debug("{} Response has status {} - which is a Client Error", SCRIPT_NAME, response.status)
+            var errorResponse = response.entity.getJson();
```

Comment:
You need to use getJsonAsync as getJson is a blocking call

Reply:
I started out using getJsonAsync, but had problems returning the response... so copied the pattern in ProcessRegistration.groovy